src/fswAlgorithms/sensorInterfaces/scanningInstrumentController/_UnitTest/test_scanningInstrumentController.py

```diff
@@ -0,0 +1,152 @@
+    """
+    
+    [testResults, testMessage] = scanningInstrumentControllerTestFunction(att_limit, att_mag, use_rate_limit, rate_limit, omega_mag, deviceStatus, controlStatus, expected_result)
+    assert testResults < 1, testMessage
+
+
+def scanningInstrumentControllerTestFunction(att_limit = 0.1, att_mag = 0.1, use_rate_limit=1, rate_limit=0.01, omega_mag=0.001, deviceStatus=None, controlStatus=None, expected_result=None):
+    """Test method"""
+    testFailCount = 0
+    testMessages = []
+    unitTaskName = "unitTask"
+    unitProcessName = "TestProcess"
+    bskLogging.setDefaultLogLevel(bskLogging.BSK_WARNING)
+
+    unitTestSim = SimulationBaseClass.SimBaseClass()
+    testProcessRate = macros.sec2nano(1.0)
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+    # setup module to be tested
+    module = scanningInstrumentController.scanningInstrumentController()
+    module.ModelTag = "scanningInstrumentControllerTag"
+    unitTestSim.AddModelToTask(unitTaskName, module)
+
+    #Initializing the test module configuration data
+    module.useRateTolerance = use_rate_limit              # rate limit enabled
+    module.rateErrTolerance = rate_limit                  # rate limit
+    module.attErrTolerance = att_limit                    # attitude error tolerance
+
+    # Configure blank module input messages
+    accessInMsgData = messaging.AccessMsgPayload()
+    accessInMsgData.hasAccess = 1                 # set access to true
+    accessInMsg = messaging.AccessMsg().write(accessInMsgData)
+
+    attGuidInMsgData = messaging.AttGuidMsgPayload()
+    attGuidInMsgData.sigma_BR = [att_mag, 0, 0]
+    attGuidInMsgData.omega_BR_B = [omega_mag, 0.0, 0.0]
+    attGuidInMsg = messaging.AttGuidMsg().write(attGuidInMsgData)
+
+    if deviceStatus is not None:
+        deviceStatusInMsgData = messaging.DeviceStatusMsgPayload()
+        deviceStatusInMsgData.deviceStatus = deviceStatus
+        deviceStatusInMsg = messaging.DeviceStatusMsg().write(deviceStatusInMsgData)
+        module.deviceStatusInMsg.subscribeTo(deviceStatusInMsg)
+
+    if controlStatus is not None:
+        module.controllerStatus = controlStatus
+
+    # subscribe input messages to module
+    module.accessInMsg.subscribeTo(accessInMsg)
+    module.attGuidInMsg.subscribeTo(attGuidInMsg)
+
+    # setup output message recorder objects
+    deviceCmdOutMsgRec = module.deviceCmdOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, deviceCmdOutMsgRec)
+
+    unitTestSim.InitializeSimulation()
+    unitTestSim.ConfigureStopTime(macros.sec2nano(1.0))
+    unitTestSim.ExecuteSimulation()
+
+    unitTestSim.ConfigureStopTime(macros.sec2nano(2.0))
+    unitTestSim.ExecuteSimulation()
+
+    # pull module data and make sure it is correct
+    if not unitTestSupport.isArrayEqual(deviceCmdOutMsgRec.deviceCmd, expected_result, len(expected_result), 1e-12):
```

Comment:
:large_orange_diamond: I recommed using [Numpy's testing support](https://numpy.org/doc/stable/reference/routines.testing.html) features. They are more robust and feature rich than the helpers we wrote 7 years ago.

Also the `testFailCount` and `testMessages` aspects are redundant and are doing an end run around Pytest's ability to collect and report testing results. I realize that the template and instructions still recommend this method so of course feel free to keep doing that. Just calling it out in case you might be interested.

Reply:
I updated the test to use Numpy's testing support as recommended.